IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -19,207 +19,273 @@ import AWSMobileClient
+            if let data = try? Data(contentsOf: URL(fileURLWithPath: myImages[0])) {
+                DispatchQueue.main.async {
+                    self.logTextView.text = "found identity \(myImages[0]), importing..."
+                }
+                if AWSIoTManager.importIdentity( fromPKCS12Data: data, passPhrase:"", certificateId:myImages[0]) {
+                    // Set the certificate ID and ARN values to indicate that we have imported
+                    // our identity from the PKCS12 file in the bundle.
+                    defaults.set(myImages[0], forKey:"certificateId")
+                    defaults.set("from-bundle", forKey:"certificateArn")
+                    DispatchQueue.main.async {
+                        self.logTextView.text = "Using certificate: \(myImages[0]))"
+                        self.iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:myImages[0], statusCallback: self.mqttEventCallback)
+                    }
+                }
             }
         }
-        else
-        {
-            activityIndicatorView.startAnimating()
-            logTextView.text = "Disconnecting..."
+        let certificateId = defaults.string( forKey: "certificateId")
+        return certificateId
+    }
 
-            DispatchQueue.global(qos: DispatchQoS.QoSClass.default).async {
-                self.iotDataManager.disconnect();
+    func createCertificateIdAndStoreinNSUserDefaults(onSuccess:  @escaping (String)->Void,
+                                                     onFailure: @escaping (Error) -> Void) {
+        let defaults = UserDefaults.standard
+        // Now create and store the certificate ID in NSUserDefaults
+        let csrDictionary = [ "commonName": CertificateSigningRequestCommonName,
+                              "countryName": CertificateSigningRequestCountryName,
+                              "organizationName": CertificateSigningRequestOrganizationName,
+                              "organizationalUnitName": CertificateSigningRequestOrganizationalUnitName]
+        
+        self.iotManager.createKeysAndCertificate(fromCsr: csrDictionary, callback: {  (responseOptional) -> Void in
+            if let response = responseOptional {
+                defaults.set(response.certificateId, forKey:"certificateId")
+                defaults.set(response.certificateArn, forKey:"certificateArn")
+                let certificateId = response.certificateId
+                print("response: [\(String(describing: response))]")
+                
+                let attachPrincipalPolicyRequest = AWSIoTAttachPrincipalPolicyRequest()
+                attachPrincipalPolicyRequest?.policyName = POLICY_NAME
+                attachPrincipalPolicyRequest?.principal = response.certificateArn
+                
+                // Attach the policy to the certificate
+                self.iot.attachPrincipalPolicy(attachPrincipalPolicyRequest!).continueWith (block: { (task) -> AnyObject? in
```

Comment:
let's move the force-unwrap to the creation of `attachPrincipalPolicyRequest` and comment it with a note that it's safe to force-unwrap because the request constructor doesn't specify nullability in the Objective-c model (I assume that's why it needs to be unwrapped)

Reply:
i'm not sure what you are suggesting here.  Are you suggesting to update the SDK?